ext/Pod-Html/lib/Pod/Html.pm

```diff
@@ -239,237 +238,233 @@ This program is distributed under the Artistic License.
         # be used throughout.
-        #$globals->{Htmlfileurl} = "$globals->{Htmldir}/" . substr( $globals->{Htmlfile}, length( $globals->{Htmldir} ) + 1);
-        # Is the above not just "$globals->{Htmlfileurl} = $globals->{Htmlfile}"?
-        $globals->{Htmlfileurl} = unixify($globals->{Htmlfile});
-
+        #$self->{Htmlfileurl} = "$self->{Htmldir}/" . substr( $self->{Htmlfile}, length( $self->{Htmldir} ) + 1);
+        # Is the above not just "$self->{Htmlfileurl} = $self->{Htmlfile}"?
+        $self->{Htmlfileurl} = unixify($self->{Htmlfile});
     }
-    return $globals;
+    return $self;
 }
 
 sub generate_cache {
-    my ($globals, $Pagesref) = @_;
+    my $self = shift;
     my $pwd = getcwd();
-    chdir($globals->{Podroot}) ||
-        die "$0: error changing to directory $globals->{Podroot}: $!\n";
+    chdir($self->{Podroot}) ||
+        die "$0: error changing to directory $self->{Podroot}: $!\n";
 
     # find all pod modules/pages in podpath, store in %Pages
     # - inc(0): do not prepend directories in @INC to search list;
-    #     limit search to those in @{$globals->{Podpath}}
+    #     limit search to those in @{$self->{Podpath}}
     # - verbose: report (via 'warn') what search is doing
     # - laborious: to allow '.' in dirnames (e.g., /usr/share/perl/5.14.1)
-    # - callback: used to remove Podroot and extension from each file
     # - recurse: go into subdirectories
     # - survey: search for POD files in PodPath
     my ($name2path, $path2name) = 
-        Pod::Simple::Search->new->inc(0)->verbose($globals->{Verbose})->laborious(1)
-        ->callback(\&_save_page)->recurse($globals->{Recurse})->survey(@{$globals->{Podpath}});
-    #print STDERR Data::Dumper::Dumper($name2path, $path2name) if ($globals->{Verbose});
+        Pod::Simple::Search->new->inc(0)->verbose($self->{Verbose})->laborious(1)
+        ->recurse($self->{Recurse})->survey(@{$self->{Podpath}});
+    # remove Podroot and extension from each file
+    for my $k (keys %{$name2path}) {
+        $self->{Pages}{$k} = _transform($self, $name2path->{$k});
+    }
 
     chdir($pwd) || die "$0: error changing to directory $pwd: $!\n";
 
     # cache the directory list for later use
-    warn "caching directories for later use\n" if $globals->{Verbose};
-    open my $cache, '>', $globals->{Dircache}
-        or die "$0: error open $globals->{Dircache} for writing: $!\n";
+    warn "caching directories for later use\n" if $self->{Verbose};
+    open my $cache, '>', $self->{Dircache}
+        or die "$0: error open $self->{Dircache} for writing: $!\n";
 
-    print $cache join(":", @{$globals->{Podpath}}) . "\n$globals->{Podroot}\n";
-    my $_updirs_only = ($globals->{Podroot} =~ /\.\./) && !($globals->{Podroot} =~ /[^\.\\\/]/);
-    foreach my $key (keys %{$Pagesref}) {
+    print $cache join(":", @{$self->{Podpath}}) . "\n$self->{Podroot}\n";
+    my $_updirs_only = ($self->{Podroot} =~ /\.\./) && !($self->{Podroot} =~ /[^\.\\\/]/);
```

Comment:
```suggestion
    my $_updirs_only = $self->{Podroot} =~ /\.\./ && $self->{Podroot} !~ /[^\.\\\/]/;
```

These kinds of lines can be tricky, so maybe you will not agree that this is a little better.  But if you agree, please feel free to accept this suggestion.

Reply:
> These kinds of lines can be tricky, so maybe you will not agree that this is a little better. But if you agree, please feel free to accept this suggestion.

Again, following the philosophy of "minimum number of changes needed to get us from Pod-Html as it appears in 5.34.0 to an OO-ified module," I recommend deferring this change until after this p.r. is in blead.